File: problem004/solution.py

def isPalindrome(nb):
  return str(nb) == str(nb)[::-1]

def largestPalindrome(dimension):
  limit = int("9" * dimension) + 1
  numbers = [str(x) for x in range(limit)]
  digitNb = list(filter(lambda x: len(str(x)) == dimension, numbers))
  digitNb = [int(x) for x in digitNb]
  palindromes = [x * y for x in digitNb for y in digitNb if isPalindrome(x * y)]
  
  return max(palindromes)

print(largestPalindrome(3)) # 906609

File: problem004/solution.js
const _ = require('lodash')
const isPalindrome = str => String(str) === [...String(str)].reverse().join('')


const largestPalindrome = dimension => {
    const limit = Number("9".repeat(dimension)) + 1
    const fromN = "1" + "0".repeat(dimension - 1)
    const numbers = _.range(fromN, limit)

    let bigPalindrome = 0
    // I would be curious to see the result with Array.reduce() method
    for (x of numbers)
        for (y of numbers)
            if (isPalindrome(x * y) && x * y > bigPalindrome)
                bigPalindrome = x * y

    return bigPalindrome
}

console.log(largestPalindrome(3)) // 906609